data/onPostBuild/generateMarkdown.ts

```diff
@@ -0,0 +1,210 @@
+const convertToMarkdown = (htmlContent: string): string => {
+  const turndownService = createTurndownService();
+
+  // Convert HTML to Markdown
+  const markdown = turndownService.turndown(htmlContent);
+
+  // Clean up excessive newlines
+  const cleanedMarkdown = markdown.replace(/\n{3,}/g, '\n\n');
+
+  return cleanedMarkdown;
+};
+
+// Write markdown file
+const writeMarkdownFile = (outputPath: string, content: string, reporter: any) => {
+  try {
+    // Ensure directory exists
+    const dir = path.dirname(outputPath);
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir, { recursive: true });
+    }
+
+    fs.writeFileSync(outputPath, content, 'utf8');
+    return true;
+  } catch (error) {
+    reporter.error(`${REPORTER_PREFIX} Error writing markdown file ${outputPath}:`, error as Error);
+    return false;
+  }
+};
+
+export const onPostBuild: GatsbyNode['onPostBuild'] = async ({ graphql, reporter, basePath }) => {
+  const publicDir = path.join(process.cwd(), 'public');
+  const docsDir = path.join(publicDir, 'docs');
+
+  // Find all index.html files in the public/docs directory
+  const htmlFiles = await fastGlob('**/index.html', {
+    cwd: docsDir,
+    absolute: false,
+  });
+
+  reporter.info(`${REPORTER_PREFIX} Found ${htmlFiles.length} HTML files to process`);
+
+  // Process all HTML files and extract metadata
+  const allPages = htmlFiles
+    .map((htmlFile) => {
+      const htmlPath = path.join(docsDir, htmlFile);
+      const html = fs.readFileSync(htmlPath, 'utf8');
+
+      // Skip redirect pages
+      if (html.length < 1000 && html.includes('window.location.href')) {
+        return null;
+      }
+
+      // Extract slug from file path (remove /index.html)
+      const slug = htmlFile.replace(/\/index\.html$/, '').replace(/^\.\//, '');
+
+      // Extract title and description from HTML meta tags
+      const $ = cheerio.load(html);
+      const title =
+        $('meta[property="og:title"]').attr('content') ||
+        $('meta[name="twitter:title"]').attr('content') ||
+        $('title').text() ||
+        'Untitled';
+      const description =
+        $('meta[name="description"]').attr('content') || $('meta[property="og:description"]').attr('content') || '';
+
+      return {
+        slug: slug || '.',
+        title,
+        description,
+      };
+    })
+    .filter((page) => page !== null) as { slug: string; title: string; description: string }[];
+
+  reporter.info(`${REPORTER_PREFIX} Processing ${allPages.length} content pages`);
+
+  let successCount = 0;
+  let failCount = 0;
+
+  for (const page of allPages) {
+    const { slug, title, description } = page;
+
+    // Determine the HTML file path
+    const htmlPath = path.join(publicDir, 'docs', slug, 'index.html');
+
+    // Extract main content from HTML
+    const htmlContent = extractMainContent(htmlPath);
+
+    if (!htmlContent) {
+      reporter.warn(`${REPORTER_PREFIX} Could not extract content for ${slug}`);
+      failCount++;
```

Comment:
Is this a failure? The asset is not generated indeed but there's a difference between "this failed" and "the content has been skipped"

Reply:
I think it is a failure because we're assuming all pages do have content (your point above). So I think this is valid, unless we have conditional logic for when we show the View Markdown option